ruleset.xml

```diff
@@ -1,5 +1,137 @@
+
+    <!-- Arrays -->
+    <rule ref="Squiz.Arrays.ArrayBracketSpacing" />
+
+    <!-- Classes -->
+    <rule ref="../TYPO3SniffPool/Sniffs/Classes/LowercaseClassKeywordsSniff.php" />
+    <rule ref="Squiz.Classes.SelfMemberReference" />
+    <rule ref="Generic.Classes.DuplicateClassName" />
+
+    <!-- CodeAnalysis -->
+    <rule ref="Generic.CodeAnalysis.ForLoopWithTestFunctionCall" />
+    <rule ref="Generic.CodeAnalysis.UnconditionalIfStatement" />
+    <rule ref="Generic.CodeAnalysis.UnnecessaryFinalModifier" />
+    <rule ref="Generic.CodeAnalysis.EmptyStatement" />
+
+    <!-- Commenting -->
+    <rule ref="PEAR.Commenting.InlineComment" />
+    <rule ref="../TYPO3SniffPool/Sniffs/Commenting/DoubleSlashCommentsInNewLineSniff.php" />
+    <rule ref="../TYPO3SniffPool/Sniffs/Commenting/ValidCommentLineLengthSniff.php" />
+    <rule ref="../TYPO3SniffPool/Sniffs/Commenting/DocCommentSniff.php" />
+    <rule ref="../TYPO3SniffPool/Sniffs/Commenting/FileCommentSniff.php" />
+    <rule ref="TYPO3SniffPool.Commenting.FileComment">
+        <exclude-pattern>ext_emconf.php</exclude-pattern>
+        <exclude-pattern>ext_tables.php</exclude-pattern>
+    </rule>
+    <rule ref="../TYPO3SniffPool/Sniffs/Commenting/ClassCommentSniff.php" />
+    <rule ref="../TYPO3SniffPool/Sniffs/Commenting/FunctionCommentSniff.php" />
+    <rule ref="../TYPO3SniffPool/Sniffs/Commenting/NoAuthorAnnotationInFunctionDocCommentSniff.php" />
+    <rule ref="Squiz.Commenting.DocCommentAlignment" />
+
+    <!-- Control structures -->
+    <rule ref="Generic.ControlStructures.InlineControlStructure" />
+    <rule ref="Squiz.ControlStructures.ControlSignature" />
+    <rule ref="../TYPO3SniffPool/Sniffs/ControlStructures/DisallowElseIfConstructSniff.php" />
+    <rule ref="../TYPO3SniffPool/Sniffs/ControlStructures/AlignedBreakStatementSniff.php" />
+    <rule ref="../TYPO3SniffPool/Sniffs/ControlStructures/DisallowEachInLoopConditionSniff.php" />
+    <rule ref="../TYPO3SniffPool/Sniffs/ControlStructures/UnusedVariableInForEachLoopSniff.php" />
+    <rule ref="../TYPO3SniffPool/Sniffs/ControlStructures/ExtraBracesByAssignmentInLoopSniff.php" />
+    <rule ref="../TYPO3SniffPool/Sniffs/ControlStructures/SwitchDeclarationSniff.php" />
+    <rule ref="../TYPO3SniffPool/Sniffs/ControlStructures/TernaryConditionalOperatorSniff.php" />
+
+    <!-- Debug -->
+    <rule ref="../TYPO3SniffPool/Sniffs/Debug/DebugCodeSniff.php">
+        <exclude-pattern>*typo3/sysext/core/Classes/Utility/DebugUtility.php</exclude-pattern>
+    </rule>
+
+    <!-- Files -->
+    <rule ref="Generic.Files.OneClassPerFile">
+        <exclude-pattern>*typo3/sysext/core/Migrations/Code/LegacyClassesForIde.php</exclude-pattern>
+    </rule>
+    <rule ref="Generic.Files.OneInterfacePerFile" />
+    <rule ref="Generic.Files.ByteOrderMark" />
+    <rule ref="Generic.Files.LineEndings" />
+    <rule ref="Generic.Files.LineLength">
+        <properties>
+            <property name="lineLimit" value="130"/>
```

Comment:
Looks a bit long to me, should we use that high values?

Reply:
I've removed it and use the default of the PSR-2 standard:
```
<!-- The soft limit on line length MUST be 120 characters; automated style checkers MUST warn but MUST NOT error at the soft limit. -->
 <rule ref="Generic.Files.LineLength">
  <properties>
   <property name="lineLimit" value="120"/>
   <property name="absoluteLineLimit" value="0"/>
  </properties>
 </rule>
```